src/Server/Completion/TokenContainer.php

```diff
@@ -0,0 +1,80 @@
+<?php
+namespace LanguageServer\Server\Completion;
+
+use LanguageServer\Protocol\Position;
+use LanguageServer\Protocol\Range;
+
+class TokenContainer
+{
+
+    const T_SINGLE_CHAR = 10000;
+
+    /**
+     *
+     * @var \LanguageServer\Server\Completion\Token[]
+     */
+    private $tokens;
+
+    public function __construct(array $tokens)
```

Comment:
Is it needed to work with the raw PHP tokens? Could you comment on the limitations of PHPParser in that regard? The error recovery should be pretty good by now: https://github.com/nikic/PHP-Parser/issues/262
I would like to work with the AST as much as possible (and if it is not capable, consider contributions).